### PR TITLE
Changed /Samples/DecoderSample to target .NET Core 3.1

### PR DESCRIPTION
**Pull request by TechPreacher**
- Updated /Samples/DecoderSample to target .NET Core 3.1
- Updated Newtonsoft.Json nuget package to 12.0.3
- Updated dockerfiles to use mcr.microsoft.com/dotnet/core/aspnet:3.1-buster-slim

### CODE REVIEW

File: Samples/DecoderSample/appsettings.Development.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-      "Default": "Debug",
-      "System": "Information",
-      "Microsoft": "Information"
-    }
-  }
-}
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
```
Comment (Mandur) [inline Samples/DecoderSample/appsettings.Development.json:4]: Is it wanted that we change the default value here?

Comment (TechPreacher) [reply Samples/DecoderSample/appsettings.Development.json]: Good point. I changed it back to the previous, original values.

File: Samples/DecoderSample/SensorDecoderModule.sln [hunk 2]
```diff
@@ -1,25 +1,25 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28010.2050
+# Visual Studio Version 16
```
Comment (fbeltrao) [inline Samples/DecoderSample/SensorDecoderModule.sln:3]: I think we don't need to change the solution version.

File: Samples/DecoderSample/ReadMe.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ You should see the result as JSON string.
 
 ![Decoder Sample - Debugging on localhost](/Docs/Pictures/decodersample-debugging.png)
 
-When running the solution in a container, the [Kestrel webserver](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-2.1) from .NET Core uses the HTTP default port 80 of the container and does not need to bind it to a port on the host machine as Docker allows for container-to-container communication. IoT Edge automatically creates the required [Docker Network Bridge](https://docs.docker.com/network/bridge/).
+When running the solution in a container, the [Kestrel webserver](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-3.1) from .NET Core uses the HTTP default port 80 of the container and does not need to bind it to a port on the host machine as Docker allows for container-to-container communication. IoT Edge automatically creates the required [Docker Network Bridge](https://docs.docker.com/network/bridge/).
```
Comment (fbeltrao) [inline Samples/DecoderSample/ReadMe.md:62]: nit: if we provide the documentation URL without the culture (en-us) people will get it in their main language.

Comment (TechPreacher) [reply Samples/DecoderSample/ReadMe.md]: Changed. Thanks for the hint.

File: Samples/DecoderSample/SensorDecoderModule.sln (hunk 2 above)
Comment (TechPreacher) [reply Samples/DecoderSample/SensorDecoderModule.sln]: Changed back.

**Pull request merged**